lib/http/timeout/null.rb

```diff
@@ -56,19 +56,31 @@ def write(data)
       private
 
       # Retry reading
-      def rescue_readable(timeout = read_timeout)
+      def rescue_readable(klass, timeout = read_timeout)
         yield
       rescue IO::WaitReadable
         retry if @socket.to_io.wait_readable(timeout)
-        raise TimeoutError, "Read timed out after #{timeout} seconds"
+        raise klass, "#{error_name_for_class(klass)} timed out after #{timeout} seconds"
       end
 
       # Retry writing
-      def rescue_writable(timeout = write_timeout)
+      def rescue_writable(klass, timeout = write_timeout)
         yield
       rescue IO::WaitWritable
         retry if @socket.to_io.wait_writable(timeout)
-        raise TimeoutError, "Write timed out after #{timeout} seconds"
+        raise klass, "#{error_name_for_class(klass)} timed out after #{timeout} seconds"
+      end
+
+      def error_name_for_class(klass)
+        if klass == ConnectTimeoutError
+          "Connect"
+        elsif klass == ReadTimeoutError
+          "Read"
+        elsif klass == WriteTimeoutError
+          "Write"
+        else
+          raise "Unknown timeout error type: #{klass}"
+        end
```

Comment:
Because the method argument is the class itself (and not an instance of it) a `case` statement doesn't work here (it uses `===`). Any ideas to clean this up?

Reply:
You can use case on klass.name for example.

Reply:
Using strings is bad in this case, however. I would suggest using a constant hash for this mapping.

Reply:
Yeah, that's much better. Changed, thanks :v: